src/GovUk.Education.ExploreEducationStatistics.Content.Model/NamingUtils.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using static System.Char;
+using static System.String;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Model
+{
+    public static class NamingUtils
+    {
+        private static readonly Regex YearRegex = new Regex(@"^([0-9]{4})?$");
+
+        public static string SlugFromTitle(string title)
+        {
+            var replaceNonAlphaNumericWithSpace = ReplaceNonAlphaNumericWithSpaceAndTrim(title);
+            var toLower = new string(replaceNonAlphaNumericWithSpace.Select(ToLower).ToArray());
+            var removeMultipleSpaces = Regex.Replace(toLower, @"\s+", " ");
+            var replaceSpaces = new string(removeMultipleSpaces.Select(c => IsWhiteSpace(c) ? '-' : c).ToArray());
+            return replaceSpaces;
+        }
+
+        private static string ReplaceNonAlphaNumericWithSpaceAndTrim(string s)
+        {
+            return new string(s.Select(ReplaceNonAlphaNumericWithSpace).ToArray()).Trim();
+        }
+
+
+        private static Func<char, char> ReplaceNonAlphaNumericWithSpace =>
+            c => IsLetter(c) || IsWhiteSpace(c) || IsDigit(c) ? c : ' ';
+
+        public static string ReleaseYearTitle(string year, TimeIdentifier coverage)
```

Comment:
Possible overlap here with a mechanism already written to format a TimeIdentifier and year.

See method `GovUk.Education.ExploreEducationStatistics.Data.Api.Services.TimePeriodLabelFormatter.Format`.

Each TimeIdentifier is annotated with a formatting convention.
See `TimePeriodLabelFormat` for the possible formats and a description of each of them.

Run the following test if you want examples of what it outputs (the test is currently skipped but when enabled it dumps a whole load of examples to the console)
`GovUk.Education.ExploreEducationStatistics.Data.Api.Tests.Services.TimePeriodLabelFormatterTests.Debug`

Reply:
Okay - so I'm now taking the year part from the formatter - but the rest remains in the release object. There might be some mileage in future rationalising some of this further